contracts/src/CrossChainApplications/NativeTokenBridge/INativeTokenDestination.sol

```diff
@@ -0,0 +1,47 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+/**
+ * @dev Interface that describes functionalities for a cross-chain native token bridge.
+ */
+interface INativeTokenDestination {
+    error InvalidTeleporterMessengerAddress();
+    error InvalidRecipientAddress();
+    error InvalidSourceChain();
+    error InvalidPartnerContractAddress();
+    error CannotBridgeTokenWithinSameChain();
+    error Unauthorized();
+    error InsufficientPayment();
+    error InvalidSourceContractAddress();
+    error InsufficientAdjustedAmount(uint256 adjustedAmount, uint256 feeAmount);
+
+    /**
+     * @dev Emitted when tokens are burned in this bridge contract to be bridged to the source chain.
+     */
+    event TransferToSource(
```

Comment:
Is there a reason to have the `teleporterMessageID` be an indexed parameter here? I think the sender and recipient would make more sense to be able to search by. Something like:

```
    event TransferToSource(
        address indexed sender,
        address indexed recipient,
        uint256 amount,
        address feeContractAddress        
        uint256 feeAmount,
        uint256 teleporterMessageID
    );
```